gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-edit-page/api-documentation-v4-edit-page.component.spec.ts

```diff
@@ -460,6 +460,177 @@ describe('ApiDocumentationV4EditPageComponent', () => {
           expect(fixture.componentInstance.form.getRawValue().content).toEqual('# Markdown content');
         });
       });
+
+      describe('step 3 - Import from URL ', () => {
+        let harness: ApiDocumentationV4EditPageHarness;
+        const openApiUrl = 'https://openapi.yml';
+        const http = 'HTTP';
+        const pageName = 'New page';
+        const emptyUrlSaveErrorMessage = 'Cannot save without a url';
+        const emptyUrlPublishErrorMessage = 'Cannot publish with empty URL';
+        beforeEach(async () => {
+          harness = await TestbedHarnessEnvironment.harnessForFixture(fixture, ApiDocumentationV4EditPageHarness);
+          await harness.setName(pageName);
+
+          let nextBtn = await harness.getNextButton();
+          expect(await nextBtn.isDisabled()).toEqual(false);
+          await nextBtn.click();
+
+          await harness.selectSource(http);
+
+          const req = httpTestingController.expectOne({
+            method: 'GET',
+            url: `${CONSTANTS_TESTING.env.baseURL}/fetchers?expand=schema`,
+          });
+
+          req.flush([
+            {
+              id: 'http-fetcher',
+              name: http,
+              description: 'The Gravitee.IO Parent POM provides common settings for all Gravitee components.',
+              version: '2.0.1',
+              schema:
+                '{"type": "object","title": "http","properties": {"url": {"title": "URL","description": "Url to the file you want to fetch","type": "string"}}}',
+            },
+          ]);
+
+          fixture.detectChanges();
+
+          nextBtn = await harness.getNextButton();
+          expect(await nextBtn.isDisabled()).toEqual(false);
+          await nextBtn.click();
+        });
+
+        it('should show http url', async () => {
+          const httpUrlInput = await harness.getHttpUrlHarness();
+          await httpUrlInput.setValue(openApiUrl);
+          expect(fixture.componentInstance.form.getRawValue().source).toEqual(http);
```

Comment:
You're testing here that the user has selected `"HTTP"` as a source. I believe that you want to rather test that the `httpUrlInput` is in the form?

Reply:
Done